docs/modules/ROOT/pages/references/style-guide.adoc

```diff
@@ -0,0 +1,273 @@
+
+http://creativecommons.org/licenses/by-nc-sa/4.0/[image:https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png[Creative Commons License]] +
+This work is licensed under a http://creativecommons.org/licenses/by-nc-sa/4.0/[Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License].
+
+== General Style
+
+* Encode text files with UTF-8.
+* Use Unix line endings (lf).
+* Always end a file with a single empty line.
+  This doesn't apply to empty files (for example `.gitkeep`).
+* Use two spaces for indentation except for `Makefile` where a single tab is to be used.
+* Trim whitespaces from the end of a line.
+
+[NOTE]
+====
+Repositories of Project Syn provide an https://editorconfig.org/[EditorConfig].
+When using a https://editorconfig.org/#download[supported editor], space types and number of spaces will automatically be done correctly.
+====
+
+== Asciidoc Style
+
+* Don't split senctences across lines but keep them on one single line.
+* Start a new line for each and every sentence.
+* Follow the https://docs.microsoft.com/en-us/style-guide/welcome/[Microsoft Writing Style Guide].
+
+[NOTE]
+====
+The Microsoft Writing Style is being enforced with https://github.com/errata-ai/vale[Vale].
+The implementation used can be found at https://github.com/errata-ai/Microsoft[errata-ai/Microsoft].
+====
+
+== Jsonnet Style
+
+== Syntactic Style
+
+=== Auto Formatting
+
+Use `jsonnet fmt` to format files. This will fix basic style errors.
+
+=== Variable Declaration
+
+* Variables should be named in `camelCase` style, and should have self-evident names.
++
+[source,jsonnet]
+----
+local serverPort = 1000;
+local clientPort = 2000;
+----
+
+* Prefer `local` to `::` syntax for private/local variables.
+  Unlike `::`, variables defined with `local` can't be overridden by children, nor accessed by other files.
++
+[source,jsonnet]
+----
+{
+  // CORRECT
+  local myVariable = 3,
+  result: myVariable + 1,
+
+  // INCORRECT
+  myVariable:: 3,
+  result: $.myVariable + 1,
+}
+----
+
+=== Line Length
+
+* Limit lines to 120 characters.
+* The only exceptions are import statements and URLs (although even for those, try to keep them under 120 chars).
+
+=== Spacing and Indentation
+
+* Put one space before and after operators.
++
+[source,jsonnet]
+----
+local c = a + b;
+----
+
+* Put one space after commas.
++
+[source,jsonnet]
+----
+["a", "b", "c"] // CORRECT
+
+["a","b","c"] // INCORRECT
+----
+
+* Put one space after colons.
++
+[source,jsonnet]
+----
+{
+  // CORRECT
+  foo:: "bar",
+  baz: "taz",
+  { hello: "world" },
+
+  // INCORRECT
+  foo :: "bar",
+  baz:"taz",
+  { hello : "world" },
+}
+----
+
+* Put one space or line brake after opening curly brackets and before closing curly brackets.
++
+[source,jsonnet]
+----
+// CORRECT
+local foo = { hello: "world" };
+local bar = {
+  hello: "world",
+};
+
+// INCORRECT
+local baz = {hello: "world"};
+----
+
```

Comment:
Probably needs a section on arrays as well (no space after `[` and before `]`), otherwise `jsonnet fmt` will complain with the default settings. Or we could consider turning on `--pad-arrays` for `jsonnet fmt`, which would make single-line arrays follow the same style as single-line objects. Personally I prefer padded arrays.

Also (as I just figured out while playing with `jsonnet fmt`) no trailing commas in single-line objects or arrays